snapwell/wellpath.py

```diff
@@ -12,13 +12,30 @@
 #  See the GNU General Public License at <http://www.gnu.org/licenses/gpl.html>
 #  for more details.
 
+import logging
+from functools import wraps
 from os.path import exists
 
 from .snap_utils import Inf, close, finiteFloat
 from .snap_utils import read_next_tokenline as token
 from .snap_utils import tryFloat
 
 
+def takes_stream(i, mode):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            if len(args) > i and args[i] is not None and isinstance(args[i], str):
```

Comment:
This will fail for `pathlib.Path` objects which can be opened. Use `isinstances(args[i], (str, pathlib.Path))`